float64 x
float64 y
float64 z
float64 x_
float64 y_
float64 z_
float64 w_


